### PR TITLE
Command send options (user, channel, dynamic and static) as options and not only values.

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
Command send options as options (pairs label/value) instead of just the values.

For dynamic options, the label we are sending is blank, due that the information about dynamic options resides in the app, and it would imply one extra fetch to the app to get again the possible options.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-33418

### CODE REVIEW

**Review by catalintomai — APPROVED** [review]

File: components/suggestion/command_provider/app_command_parser.ts [hunk 1]
```diff
@@ -610,6 +613,39 @@ export class AppCommandParser {
...
+                break;
+            case AppFieldTypes.STATIC_SELECT:
+                values[f.name] = f.options?.find((o) => (o.value === values[f.name]));
+                break;
+            case AppFieldTypes.USER: {
+                let userName = values[f.name] as string;
+                if (userName[0] === '@') {
+                    userName = userName.substr(1);
+                }
+                const user = getUserByUsername(this.store.getState(), userName);
```
Comment (mickmister) [inline components/suggestion/command_provider/app_command_parser.ts:633]: What happens when this user has not been loaded from the server yet? e.g. We just refreshed and ran the command as is. Or the username is spelled wrong or is incomplete

Comment (larkox) [reply components/suggestion/command_provider/app_command_parser.ts:633]: Current behaviour: an option {label: undefined, value: undefined} would be sent.

What do you think about this approach?
1. Try to get user from redux
2. If not found, try to get user using the getUserByUsername action, which will trigger a call to the server.
3. If not found, somehow surface the error (and return null for this call).

**Review by levb — COMMENTED** [review]
👍

**Review by levb — APPROVED** [review]

**Review by mickmister — APPROVED** [review]
Awesome job Daniel!

**Pull request merged**